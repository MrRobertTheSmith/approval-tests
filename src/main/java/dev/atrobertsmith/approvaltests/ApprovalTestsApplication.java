package dev.atrobertsmith.approvaltests;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
public class ApprovalTestsApplication {

	public static void main(String[] args) {
		SpringApplication.run(ApprovalTestsApplication.class, args);
	}

}
