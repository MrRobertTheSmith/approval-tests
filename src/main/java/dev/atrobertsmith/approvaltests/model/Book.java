package dev.atrobertsmith.approvaltests.model;

public record Book(String title, String author) { }
