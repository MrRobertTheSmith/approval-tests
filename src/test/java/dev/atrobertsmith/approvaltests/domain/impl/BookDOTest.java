package dev.atrobertsmith.approvaltests.domain.impl;

import dev.atrobertsmith.approvaltests.model.Book;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

// This test tries out the
class BookDOTest {

    private BookDO testTarget;

    @BeforeEach
    void setUp() {
        testTarget = new BookDO();
    }

    @Test
    void fetchHappyPath() {
        //GIVEN two books have been saved
        testTarget.saveBook(new Book("Cat's Cradle", "K Vonnegut"));
        testTarget.saveBook(new Book("Julian", "G Vidal"));

        //WHEN
        var result = testTarget.fetchBooks();

        //THEN
        Approvals.verify(result);
    }

    @Test
    void saveHappyPath() {
        //GIVEN two books have been saved
        //WHEN
        var result = testTarget.saveBook(new Book("Cat's Cradle", "K Vonnegut"));

        //THEN
        Approvals.verify(result);
    }

}