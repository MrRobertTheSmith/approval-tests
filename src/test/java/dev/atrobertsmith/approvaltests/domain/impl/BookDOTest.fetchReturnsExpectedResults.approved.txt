[Book[title=Cat's Cradle, author=K Vonnegut], Book[title=Julian, author=G Vidal]]